### PR TITLE
Represent TerraformBackend.Config with cty.Value

### PR DESCRIPTION
**Pull request by mr0re1**
* Add `Dict` struct to map string ket to `cty.Value` and handle YAML marshal/unmarshal;
* Use `Dict` instead of `map[string]interface{}` for `TerraformBackend.Config`.

### Submission Checklist

* [ ] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [ ] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [ ] Is unit test coverage still above 80%?
* [ ] Have you updated all applicable documentation?
* [ ] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Review by tpdownes — CHANGES_REQUESTED** [review]
- fix minor typos
- address concern about old library (if there is no alternative please confirm that this isn't tying us to old packages or old version of Golang)
- minor Qs about parsing

May not require code changes other than typos.

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,9 @@ require (
 require (
 	cloud.google.com/go/serviceusage v1.6.0
 	github.com/go-git/go-billy/v5 v5.4.1
+	github.com/google/go-cmp v0.5.9
 	github.com/googleapis/gax-go/v2 v2.8.0
+	github.com/zclconf/go-cty-debug v0.0.0-20191215020915-b22d67c1ba0b
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply go.mod:29]: Sadly none that I'm aware of. On bright side, It's only a test dependency.

File: pkg/config/dict.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+func (d *Dict) Has(k string) bool {
+	if d.m == nil {
+		return false
+	}
+	_, ok := d.m[k]
+	return ok
+}
+
+// Set adds/overrides value by key.
+// Returns refernce to Dict-self.
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply pkg/config/dict.go:51]: addressed

File: pkg/config/dict.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+func (y *yamlValue) UnmarshalYAML(n *yaml.Node) error {
+	var err error
+	switch n.Kind {
+	case yaml.ScalarNode:
+		err = y.unmarshalScalar(n)
+	case yaml.MappingNode:
+		err = y.unmarshalObject(n)
+	case yaml.SequenceNode:
+		err = y.unmarshalTuple(n)
+	default:
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply pkg/config/dict.go:89]: I've referenced `go-yaml` codebase, they don't deal with it neither (in `Decode`) `Alias` nor `Document`. 
It just works, added a test `TestYAMLDecodeWithAlias`.

**Review by tpdownes — APPROVED** [review]

**Pull request merged**